internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -136,56 +184,146 @@ func (c *cmd) Execute() (statusCode int, err error) {
+	// Save the bootstrap ACL token into json file so that it can be used later on
 	if err := c.saveBootstrapACLToken(bootstrapACLToken); err != nil {
-		return interfaces.StatusCodeExitWithError, fmt.Errorf("failed to save registry's bootstrap ACL token: %v", err)
+		return fmt.Errorf("failed to save registry's bootstrap ACL token: %v", err)
 	}
 
-	// retrieve the secretstore (Vault) token from the file produced by secretstore-setup
-	secretstoreToken, err := c.getSecretStoreTokenFromFile()
+	// create a management token and store it to a file
+	if err := c.createAndSaveManagementToken(bootstrapACLToken.SecretID); err != nil {
+		return fmt.Errorf("failed to save management token: %v", err)
+	}
+
+	return nil
+}
+
+// createEdgeXACLTokenRoles creates secret store roles that can be used for genearting registry tokens
+// via Consul secret engine API /consul/creds/[role_name] later on for all EdgeX microservices
+func (c *cmd) createEdgeXACLTokenRoles(bootstrapACLTokenID, secretstoreToken string) error {
+	edgexServicePolicy, err := c.getOrCreateRegistryPolicy(bootstrapACLTokenID, edgeXServicePolicyName, edgeXPolicyRules)
 	if err != nil {
-		return interfaces.StatusCodeExitWithError, fmt.Errorf("failed to retrieve secretstore token: %v", err)
+		return fmt.Errorf("failed to create edgex service policy: %v", err)
 	}
 
-	c.loggingClient.Info("successfully get secretstore token and configuring the registry access for secretestore")
+	roleNames := c.configuration.StageGate.Registry.ACL.GetACLRoleNames()
+	if len(roleNames) == 0 {
+		c.loggingClient.Warn("found no ACL role names defined in configuration, skip create ACL roles")
+		return nil
+	}
 
-	// configure Consul access with both Secret Store token and consul's bootstrap acl token
-	if err := c.configureConsulAccess(secretstoreToken, bootstrapACLToken.SecretID); err != nil {
-		return interfaces.StatusCodeExitWithError, fmt.Errorf("failed to configure Consul access: %v", err)
+	for _, roleName := range roleNames {
+		// create roles based on the service keys as the role names
+		// in phase 2, we are using the same policy rule for all services
+		edgexACLTokenRole := NewRegistryRole(roleName, ClientType, []Policy{
+			*edgexServicePolicy,
+			// localUse set to false as some EdgeX services may be running in a different node
+		}, false)
+
+		// fail all if any one of the role creation failed
+		if err := c.createRole(secretstoreToken, edgexACLTokenRole); err != nil {
+			return fmt.Errorf("failed to create edgex role: %v", err)
+		}
 	}
 
-	// write a sentinel file to indicate Consul ACL bootstrap is done so that we don't bootstrap ACL again,
-	// this is to avoid re-bootstrapping error and that error can cause the snap crash if restart this process
-	if err := c.writeSentinelFile(); err != nil {
-		return interfaces.StatusCodeExitWithError, fmt.Errorf("failed to write sentinel file: %v", err)
+	return nil
+}
+
+func (c *cmd) createAndSaveManagementToken(bootstrapACLTokenID string) error {
```

Comment:
When I wrote the ADR, I considered the "management token" to be the built-in global-management-token role.  This seems to be the spec for a global client token, not the management token.

Reply:
@bnevis-i   ok, do we need all the global management permissions?  Here are the rules for global-management token:
```hcl

acl = "write"
agent_prefix "" {
    policy = "write"
}
event_prefix "" {
    policy = "write"
}
key_prefix "" {
    policy = "write"
}
keyring = "write"
node_prefix "" {
    policy = "write"
}
operator = "write"
query_prefix "" {
    policy = "write"
}
service_prefix "" {
    policy = "write"
    intentions = "write"
}
session_prefix "" {
    policy = "write"
}"

```

For proxy-setup to configure, i don't know whether it needs that so many permissions or not?
Can you gives a list of permission that we actually need?

Reply:
I have not.  Suggest add tightening this down to phase 3.

Reply:
ok, I removed all these save management token codes